kolibri/plugins/coach/assets/src/modules/coreCoach/actions.js

```diff
@@ -43,7 +43,7 @@ export function resetModuleState(store, { toRoute, fromRoute }) {
     return store.dispatch('examCreation/resetExamCreationState');
   }
   const moduleName = pageNameToModuleMap[fromRoute.name];
-  if (moduleName) {
+  if (moduleName && fromRoute.name !== toRoute.name) {
```

Comment:
> Don't reset state when navigating between instances of the same page.

can you elaborate on why? I would expect this to be necessary

Reply:
Because we avoid setting core loading to true when we navigate between similar instances of the same page, this causes breakage as the state is cleared, and the component tries to read from nested state that is now undefined.

Reply:
The alternative is to fully specify the nested state in the initial state (but will still cause odd flashes as the state is cleared and then waits to load).

Reply:
(generally, it is also unnecessary on navigation from the same page to the same page, as we fully specify the state in the handler, and so any previous state is overwritten when the new state is committed).